packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -44,106 +70,72 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
       <div
-        className={ css('ms-Rating', this.props.className, {
-          ['ms-Rating--large ' + styles.rootIsLarge]: this.props.size === RatingSize.Large
-        }) }
-        role='application'
+        className={ css('ms-Rating-star') }
+        aria-label={ this.props.getAriaLabel ? this.props.getAriaLabel(this.state.rating ? this.state.rating : 0, this.props.max as number) : '' }
       >
-        <div className={ css('ms-Rating-container', styles.container) } role='radiogroup' aria-labelledby={ this.props.ariaLabelId }>
+        <FocusZone
+          direction={ FocusZoneDirection.horizontal }
+          tabIndex={ this.props.readOnly ? 0 : -1 }
+          className={ css('ms-Rating-focuszone', styles.ratingFocusZone) }
+        >
           { stars }
-        </div>
-      </div>
-    );
-  }
-
-  private _renderStar(rating: number): JSX.Element {
-    const inputId = `${this._id}-${rating}`;
-
-    return (
-      <div
-        className={ css('ms-Rating-star', styles.star, {
-          ['is-selected ' + styles.starIsSelected]: rating <= (this.state.rating as number),
-          ['is-inFocus ' + styles.starIsInFocus]: rating === this.state.focusedRating,
-          ['is-disabled ' + styles.starIsDisabled]: this.props.disabled
-        }) }
-        key={ rating }
-      >
-        <input
-          className={ css('ms-Rating-input', styles.input) }
-          type='radio'
-          name={ this._id }
-          id={ inputId }
-          value={ rating }
-          aria-labelledby={ `${this._labelId}-${rating}` }
-          disabled={ this.props.disabled }
-          checked={ rating === this.state.rating }
-          onChange={ this._onChange.bind(this, rating) }
-          onFocus={ this._onFocus.bind(this, rating) }
-          onBlur={ this._onBlur.bind(this, rating) }
-        />
-        <label className={ css('ms-Rating-label', styles.label) } htmlFor={ inputId }>
-          { this._getLabel(rating) }
-          { this._getIcon() }
-        </label>
+        </FocusZone>
       </div>
     );
   }
 
   private _onFocus(value: number, ev: React.FocusEvent<HTMLElement>): void {
     this.setState({
-      focusedRating: value
-    } as IRatingState);
-
-    if (this.props.onFocus) {
-      this.props.onFocus(ev);
-    }
-  }
-
-  private _onBlur(option: number, ev: React.FocusEvent<HTMLElement>): void {
-    this.setState({
-      focusedRating: null
-    } as IRatingState);
-
-    if (this.props.onBlur) {
-      this.props.onBlur(ev);
-    }
-  }
-
-  private _onChange(rating: number, evt: React.FormEvent<HTMLInputElement>) {
-    this.setState({
-      rating: rating
+      rating: value
     } as IRatingState);
 
     const { onChanged } = this.props;
     if (onChanged) {
-      onChanged(rating);
+      onChanged(value);
     }
   }
 
   private _getLabel(rating: number): JSX.Element {
-    const text = this.props.ariaLabelIcon || 'Star';
+    const text = this.props.ariaLabelFormat || '{0} of {1} stars selected';
```

Comment:
Can you please completely remove the hardcoded English string and make sure that empty strings are fine here? (Which requires you update the snapshot)

Reply:
This is the string that is read by screen readers. Not sure if we should keep it empty by default. Even for other components, there are default strings which are used when no user provided strings are present. Is there a way to localize the default strings?

Reply:
Yes it should not have English defaults. If weneed required strings, then create a strings interface that is required. In this case, let it be up to the caller.